File: Bella/Bella.Services/Database/Hairstyle.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Bella.Services.Database
{
    public class Hairstyle
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string Name { get; set; } = string.Empty;
        
        public byte[]? Image { get; set; }
        
        [Required]
        public decimal Price { get; set; }
        
        public bool IsActive { get; set; } = true;

        [Required]
        public int LengthId { get; set; }
        public virtual Length Length { get; set; } = null!;

        [Required]
        public int GenderId { get; set; }
        public virtual Gender Gender { get; set; } = null!;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Bella/UI/bella_hairdresser_mobile/lib/model/hairstyle.dart
import 'package:json_annotation/json_annotation.dart';

part 'hairstyle.g.dart';

@JsonSerializable()
class Hairstyle {
  final int id;
  final String name;
  final String? image;
  final double price;
  final bool isActive;
  final DateTime createdAt;
  final int lengthId;
  final String lengthName;
  final int genderId;
  final String genderName;

  Hairstyle({
    this.id = 0,
    this.name = '',
    this.image,
    this.price = 0.0,
    this.isActive = true,
    required this.createdAt,
    this.lengthId = 0,
    this.lengthName = '',
    this.genderId = 0,
    this.genderName = '',
  });

  factory Hairstyle.fromJson(Map<String, dynamic> json) =>
      _$HairstyleFromJson(json);
  Map<String, dynamic> toJson() => _$HairstyleToJson(this);
}

